app/routes/user/user_data.server.ts

```diff
@@ -0,0 +1,43 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+
+import { tables } from '@architect/functions'
+import { getUser } from '../__auth/user.server'
+
+export class UserDataServer {
+  #sub: string
+
+  private constructor(sub: string) {
+    this.#sub = sub
+  }
+
+  static async create(request: Request) {
+    const user = await getUser(request)
+    if (!user) throw new Response('not signed in', { status: 403 })
+    return new this(user.sub)
+  }
+
+  async updateUserData(username: string, association: string) {
+    const db = await tables()
+
+    await db.user_data.update({
+      Key: { sub: this.#sub },
+      UpdateExpression: 'set username = :username, association = :association',
```

Comment:
There are too many things that might be considered a "username". Let's call this field "name" or "displayName".

Also... would be good to populate the default value from the user's OpenID scope.

Reply:
Do we have that available in scope info? Or would we need to update the mappings in cognito?

Reply:
`openid` is a standard scope. Refer to the OpenID Connect documentation for the standard claims included in that scope.

Or, `console.log` the ID token.